### ISSUE TITLE
"EVL205: missing scriptencoding" can't be turned off

### ISSUE THREAD

**Issue by lcd047** [syngan/vim-vimlint#81]
To reproduce, create a file with `test.vim` contents
```vim
let g:foo = '⚠'
```
then run `:call vimlint#vimlint(expand('%'), {'quiet': 1, 'EVL205': 1})`.  Possibly related to #48.

**Comment by syngan** [comment]
@wsdjeg
Could you let me know why you want to ignore `EVL205`.

**Comment by wsdjeg** [comment]
just like this
'''vim
let g:foo = '⚠'
'''
there is no wrong ,so there is no need tell me 
and sometimes i will add some chinese in to my vimrc,also no need tell me

**Comment by wsdjeg** [comment]
also the encoding checking should baseed on my own encoding,i can see what the char is ,and without error,then checker should not show warning

**Comment by lcd047** [comment]
@wsdjeg It _is_ wrong, or at least it isn't right.  Something like this:
```vim
scriptencoding utf-8
let g:foo = '⚠'
echo split(g:foo, '\zs\ze')
```
produces different results than this:
```vim
scriptencoding cp1252
let g:foo = '⚠'
echo split(g:foo, '\zs\ze')
```

**Comment by wsdjeg** [comment]
i test this by write to file test.vim and source it ,show me two different result
![2015-09-29 16-20-05](https://cloud.githubusercontent.com/assets/13142418/10158709/703060fe-66c6-11e5-9361-1b890fa95fd3.png)
![2015-09-29 16-20-43](https://cloud.githubusercontent.com/assets/13142418/10158720/7e5bb0ca-66c6-11e5-8e9e-f82293ff6f46.png)
but how can i config my vimrc ,should i add scriptencoding utf-8 into my vimrc and before the line of ```
'⚠'
```

**Comment by wsdjeg** [comment]
thank you i think i have fix my vimrc by add the utf-8 to the first line

**Comment by wsdjeg** [comment]
by now there is no need to  ignore EVL205

**Comment by lcd047** [comment]
You need to add it at the top of your vimrc:
```vim
set nocompatible
scriptencoding utf-8
...
```
Still, not being able to disable the warning about it in `vimlint` is equally wrong.  A feature that can't be turned off is a bug.

**Comment by wsdjeg** [comment]
now the warning is desappears! thank you for help me with the issue!

**Comment by syngan** [comment]
@lcd047  thank you for your help.

>  A feature that can't be turned off is a bug.

It is not a bug. If the future worked, @wsdjeg did not recognize that `:scriptencoding` is needed.
I don't think `vimlint` has the feature for `EVL205` (and `EVL105`).

**Comment by lcd047** [comment]
There are situations when it's perfectly legitimate to disable any kind of warnings and error messages, f.i. when checking somebody else's code that is widely considered to work as intended.  This is particularly important for `EVL105`: a lot of older plugins don't bother with `g:` for config variables and the like, and have been working fine for years.

On the other hand, if disabling certain error messages in `vimlint` is not possible by design and this is actually the intended behaviour, this position should be documented.

**Comment by syngan** [comment]
I have update the document. Thank you for your comment.

**Comment by wsdjeg** [comment]
@syngan can you see this log?
https://travis-ci.org/wsdjeg/DotFiles/jobs/132992413
should I add `scriptencoding utf-8` into each file? as I know `scriptencoding` can be set the in first script, such as `~/.vim/vimrc`, I think no need add it into each file.

**Comment by tyru** [comment]
> I think no need add it into each file.

@wsdjeg No, you have to. `:scriptencoding` is only effective per one file.

**Comment by wsdjeg** [comment]
@tyru Thanks, I got it.
